internal/oc/release.go

```diff
@@ -442,21 +444,158 @@ func execute(log logrus.FieldLogger, executer executer.Executer, pullSecret stri
 	}
 }
 
-// Create a temporary file containing the ImageContentPolicySources
-func (r *release) getIcspFileFromRegistriesConfig(log logrus.FieldLogger) (string, error) {
+// whiteSpaceRE is the regular expression used to split commands into arguments, taking into account that the separator
+// can be one or multiple whilte spaces.
+var whiteSpaceRE = regexp.MustCompile(`\s+`)
 
+// getMirrorsFromRegistriesConfig returns the information needed to generate the command line flag that specifies the
+// mirrors configuration used by the 'oc' command.
+func (r *release) getMirrorsFlagFromRegistriesConfig(log logrus.FieldLogger) (result *mirrorsFlagInfo, err error) {
 	if !r.mirrorRegistriesBuilder.IsMirrorRegistriesConfigured() {
-		log.Debugf("No mirrors configured to build ICSP file")
+		log.Debugf("No mirrors configured to build mirrors file")
+		return
+	}
+
+	mirrorsRegistriesConfig, err := r.mirrorRegistriesBuilder.ExtractLocationMirrorDataFromRegistries()
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get the mirror registries needed for mirrors file")
+		return
+	}
+
+	// Check if we should use the deprecated ImageContentPolicySource object or the new ImageDigestMirrorSet. To do
+	// so we check if the 'oc adm release info' command supports the '--idms-file' flag.
+	suportsIdmsFileFlag, err := r.supportsIdmsFileFlag(log, templateReleaseInfo)
+	if err != nil {
+		return
+	}
+	var (
+		name string
+		file string
+	)
+	if suportsIdmsFileFlag {
+		name = idmsFileFlagName
+		file, err = r.getIdmsFileFromRegistriesConfig(log, mirrorsRegistriesConfig)
```

Comment:
If `file` comes back empty like [here](https://github.com/openshift/assisted-service/pull/6764/files#diff-3b8a020ef087b81b6b45d56bef6dd8d32fcf9d50dfe3d91fd369755b253cfff8R538) and error is nil, will that affect the formatting of the final [command](https://github.com/openshift/assisted-service/pull/6764/files#diff-3b8a020ef087b81b6b45d56bef6dd8d32fcf9d50dfe3d91fd369755b253cfff8R396)?

Reply:
That would result in `--icsp-file=`, without a value. But I believe that can't happen due to how the `getIdmsFileFromRegistrationConfig` is written: it can only return an empty file if the `os.CreateTemp` function returns an empty file and no error. Anyhow I can explicitly check it here if you think it is convenient.

Reply:
Added a check in this commit: https://github.com/openshift/assisted-service/pull/6764/commits/49952f2106fde101384028e215a4ccaf8ac2236a . Will squash when this is ready to merge.

Reply:
Hmm what about having a check here instead: https://github.com/openshift/assisted-service/pull/6764/files#diff-3b8a020ef087b81b6b45d56bef6dd8d32fcf9d50dfe3d91fd369755b253cfff8R515
To see if file and name are both set to a string? and return "" if not

E.g.
```
func (i *mirrorsFlagInfo) String() string {
	if i == nil {
		return ""
	}
         if i.name == "" || i.file == "" {
		return ""
	}
	return fmt.Sprintf("--%s=%s", i.name, i.file)
}
```

I'm a bit worried that returning in error will cause the function [here](https://github.com/openshift/assisted-service/pull/6764/files#diff-3b8a020ef087b81b6b45d56bef6dd8d32fcf9d50dfe3d91fd369755b253cfff8R333) to error out, which shouldn't be the case because sometimes there are no mirror registries configured and the command should still run

Reply:
No problem, done.